app/api/domain/forward_request_model.py

```diff
@@ -0,0 +1,55 @@
+from pydantic import BaseModel, field_validator, model_validator
+
+from .exception import (
+    AccessDeniedError,
+    InvalidValueError,
+    MissingValueError,
+)
+
+
+class ForwardRequest(BaseModel):
+    """A domain-level data model that encapsulates all the essential information needed to forward a clients request to an external backend system."""
+
+    application_id: str
+    forward_to: str
+    patient_nhs_number: str
+    patient_ods_code: str
+    proxy_nhs_number: str
+
+    @model_validator(mode="before")
+    def validate_required_value(cls, values: list) -> list:
+        """Validates if required value is present.
+
+        If unsuccessful will raise a MissingValueError Exception
+        """
+        required_fields = ["application_id", "patient_ods_code", "forward_to"]
+        for required_field in required_fields:
+            if required_field not in values or not values[required_field]:
+                msg = "Missing required value"
+                raise MissingValueError(msg)
+        return values
+
+    @model_validator(mode="before")
+    def validate_nhs_number(cls, values: list) -> list:
```

Comment:
Do we want validation in here e.g. number length/making sure it is just numerical etc...